### DISCUSSION TITLE
REQ: Changes for mvd_auto_track_custom cvars

### DISCUSSION THREAD

**Discussion by Gummar** [ezQuake/ezquake-source#516] [Ideas]
I've found some variables that are very close to what I wanted for them to do for mvd_auto_track_custom (%C and %c), although, instead of considering "what I did in this entire run" or "how much time I survived in this run", it uses "what I already did in this run" or "how much time I'm surviving". I want to use (%C-3)/%c) in the second specified way, with it, I will be able to see players that did at least 3 kills, but it will be focusing the player who did those kills in less time. In this case, a cvar that changes the behavior of the tracker by using all they did in a run instead of what they currently did in the run I watch, I think a cvar would be better because I don't find anything to mix variables from the first method with the second method, not sure if there is anything that would be good to mix.

%c is a time value which starts at 0, meaning that if you pick any variable and divide by it, you will get an division by 0 error in the console, in those cases, just giving 0 in the result of division will solve the problem.

The frag variables should have a damage version of them so that it is possible to consider players that causes damage but doesn't kill other players as trackeable/skilled players, also one damage variable that picks damage but only the damage that wasn't healed by the damaged player.

**Comment by Gummar** [comment]
After some thought, I found that it should be better to have 4 (16 if you a kind of variables for all variables, for example: %fa, %fb, %fc, %fd.
The second letter "a" means "I did in this run until now", "b" means "I did in this run", "c" means "I did in this match until now", "d" means "I did in this match". First letter "f" is specifying what I did do, in this case it is frags.
There also should be quad / invi variants by adding third letters, example: %faq, %fci
"%faq" should be "Frags I did in this run until now while having quad damage", "%fci" should be "Frags I did in this match until now while having invisibility", you can also use the two variables at the same time by using (%fbq+%fbi)/2.
Others variables that should be added is first kill, time since the last kill, time until next kill, last kill (returns 1 or 0). The time since the last kill can be used to give a player some points to make the viewer see the dead player dieing, instead of instantly switching views, time until next kill can be used to make the viewer switch to a player that is about to kill someone, first kill and last kill should give the player the difference in time to his first kill / last kill. Although, for better use of those variables, funcions like max(variables) and min(variables) should be added.

**Comment by jite** [comment]
I will look into this when I get time, thanks.
